app/checkout/[userId]/page.tsx

```diff
@@ -84,16 +95,28 @@ export default function CheckOut({ params }: CheckOutProps) {
             <VisitInfoForm
                 initialVisitData={visitsData[0]}
                 onSubmit={() => setShow(true)}
+                onChange={setVisitFormData}
                 submitLabel="Save and Check Out"
             />
             <Modal show={show} setShow={setShow}>
                 <h3>Are you sure you want to checkout this client?</h3>
                 <h4>{`${fullName}`}</h4>
                 <hr />
                 <div className={styles.rowContainer}>
-                    <Button onClick={() => checkOut(visitsData[0])}>
+                    <Button
+                        onClick={() =>
+                            visitFormData && checkOut(visitFormData!)
+                        }
+                    >
                         Confirm Checkout
                     </Button>
+                    <Button
+                        onClick={() =>
+                            visitFormData && saveVisit(visitFormData!)
+                        }
+                    >
+                        Save Only
+                    </Button>
```

Comment:
One thing that I did not notice in previous review. Right now the `Confirm Checkout` and `Save Only` buttons do nothing without a change in visit form. So, user cannot check out a client without modifying the form. 

Can we let user saving the current form data and checkout client without modifying it?

Reply:
Thanks for pointing out! I think we can set the initial state of visitFormData to current visitsData[0]
```suggestion
    const [visitFormData, setVisitFormData] = useState<Visit>(
        visitsData ? visitsData[0] : ({} as Visit)
    );
```